oauthenticator/github.py

```diff
@@ -191,8 +191,19 @@ async def authenticate(self, handler, data=None):
         auth_state['access_token'] = access_token
         # store the whole user model in auth_state.github_user
         auth_state['github_user'] = resp_json
-        # A public email will return in the initial query (assuming default scope).
-        # Private will not.
+        # To retrieve emails an extra call has to be made to a secondary endpoint
+        # using the access token. see https://github.com/jupyterhub/oauthenticator/issues/438
+        if 'user:email' in self.scope:
```

Comment:
i'm not sure if this is the only scope to check or support for emails.

Reply:
`email` is access to the public email address
`user:email` is access to the private email addresses (note plural)
`user` or `read:user` would imply `user:email` and perhaps some more details, so you can check for `read:user` and `user` alongside `user:email` to decide if this check should be made.

Details here: https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes